plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js

```diff
@@ -330,19 +330,34 @@ jQuery( function ( $ ) {
 	);
 
 	// Stock options.
+	function show_or_hide_stock_management_fields(
+		isStockManagementEnabled,
+		productType
+	) {
+		const stockManagementFields = $( '.stock_fields' );
+		const stockStatusField = $( '.stock_status_field' );
+
+		if ( isStockManagementEnabled ) {
+			stockManagementFields.show();
+			stockStatusField.hide();
+		} else {
+			stockManagementFields.hide();
+			// do not show stock status field if it should be hidden for the product type
+			if ( ! stockStatusField.is( '.hide_if_' + productType ) ) {
+				stockStatusField.show();
+			}
+		}
```

Comment:
What do you think about using the method `.toggle` here?

The code would look like this:
```
	stockManagementFields.toggle( isStockManagementEnabled );
	stockStatusField.toggle(
		! isStockManagementEnabled &&
			! stockStatusField.is( '.hide_if_' + productType )
	);
```

Reply:
I like it, nice suggestion!